core/src/main/java/org/commonjava/indy/core/content/DefaultDirectContentAccess.java

```diff
@@ -40,21 +47,60 @@
     @Inject
     private DownloadManager downloadManager;
 
+
     public DefaultDirectContentAccess(){}
 
-    public DefaultDirectContentAccess( DownloadManager downloadManager )
+    public DefaultDirectContentAccess( final DownloadManager downloadManager )
     {
         this.downloadManager = downloadManager;
     }
 
+
+    @Override
     public List<Transfer> retrieveAllRaw( final List<? extends ArtifactStore> stores, final String path,
                                           final EventMetadata eventMetadata )
             throws IndyWorkflowException
     {
-        final List<Transfer> txfrs = new ArrayList<Transfer>();
+        Logger logger = LoggerFactory.getLogger( getClass() );
+        ExecutorService executor = Executors.newCachedThreadPool();
+        final Map<ArtifactStore, Future<Transfer>> futureMap = new HashMap<>();
         for ( final ArtifactStore store : stores )
         {
-            final Transfer txfr = retrieveRaw( store, path, eventMetadata );
+            logger.debug( "Requesting retrieval of {} in {}", path, store );
+            final Future<Transfer> txfrFuture = executor.submit( new Callable<Transfer>()
+            {
+                @Override
+                public Transfer call() throws IndyWorkflowException
+                {
+                    logger.trace( "Retrieving {} in {}", path, store );
+                    Transfer txfr = retrieveRaw( store, path, eventMetadata );
+                    logger.trace( "Transfer {} in {} retrieved", path, store );
+                    return txfr;
+                }
+            });
+            futureMap.put( store, txfrFuture );
```

Comment:
Using ExecutorCompletionService, I think we could map the Transfer result directly to the `store` here. Then, below, we could iterate the `stores` parameter to return the Transfers in the appropriate order (taken from the group membership ordering, presumably).